app/jobs/users_birthday_job.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'sidekiq-scheduler'
+require 'date'
+
+class UsersBirthdayJob < ApplicationJob
+  queue_as :default
+
+  def perform(*_args)
+    # Rails.logger.debug User.where(birthday: '1992-02-22')
+    User.where(
+      "DATE_TRUNC('day', birthday) = DATE_TRUNC('day', CURRENT_DATE) AND
+        DATE_TRUNC('month', birthday) = DATE_TRUNC('month', CURRENT_DATE)"
+    ).find_each do |user|
+      BirthdayMailer.birthday_email(user&.email, recipient_name(user))
```

Comment:
```suggestion
      BirthdayMailer.birthday_email(user.email, user.full_name).deliver
```
поле email будет всегда у пользователя, тут не нужен блок try
recipient_name - где у тебя этот метод обьявлен ?  надо добавить метод для объекта и вызывать для пользователя

Reply:
Этот метод находится в  app->mailers->birthday_mailer.rb. Туда передаётся юзер

Reply:
путь не явный перенеси в модель